// @frontity/.ignored_mars-theme/src/model/FooterNewsCategoryModel.js
export const FooterNewsCategoryModel = {
  term_id: 225,
  name: "News Category",
  slug: "news-category",
  term_group: 0,
  term_taxonomy_id: 225,
  taxonomy: "nav_menu",
  description: "",
  parent: 0,
  count: 7,
  filter: "raw",
  items: [
    {
      ID: 70895,
      post_author: "1",
      post_date: "2019-06-18 09:41:04",
      post_date_gmt: "2019-06-18 08:41:04",
      post_content: " ",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "70895",
      to_ping: "",
      pinged: "",
      post_modified: "2019-06-18 09:42:38",
      post_modified_gmt: "2019-06-18 08:42:38",
      post_content_filtered: "",
      post_parent: 0,
      guid: "https://fastsole.co.uk/?p=70895",
      menu_order: 1,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 70895,
      menu_item_parent: "0",
      object_id: "7",
      object: "post_tag",
      type: "taxonomy",
      type_label: "Tag",
      url: "/sneaker-news/tag/nike/",
      title: "Nike",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "nike",
    },
    {
      ID: 70896,
      post_author: "1",
      post_date: "2019-06-18 09:41:06",
      post_date_gmt: "2019-06-18 08:41:06",
      post_content: " ",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "70896",
      to_ping: "",
      pinged: "",
      post_modified: "2019-06-18 09:42:40",
      post_modified_gmt: "2019-06-18 08:42:40",
      post_content_filtered: "",
      post_parent: 0,
      guid: "https://fastsole.co.uk/?p=70896",
      menu_order: 2,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 70896,
      menu_item_parent: "0",
      object_id: "29",
      object: "post_tag",
      type: "taxonomy",
      type_label: "Tag",
      url: "/sneaker-news/tag/adidas/",
      title: "adidas",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "adidas",
    },
    {
      ID: 70897,
      post_author: "1",
      post_date: "2019-06-18 09:41:06",
      post_date_gmt: "2019-06-18 08:41:06",
      post_content: " ",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "70897",
      to_ping: "",
      pinged: "",
      post_modified: "2019-06-18 09:42:40",
      post_modified_gmt: "2019-06-18 08:42:40",
      post_content_filtered: "",
      post_parent: 0,
      guid: "https://fastsole.co.uk/?p=70897",
      menu_order: 3,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 70897,
      menu_item_parent: "0",
      object_id: "22",
      object: "post_tag",
      type: "taxonomy",
      type_label: "Tag",
      url: "/sneaker-news/tag/air-jordan/",
      title: "Air Jordan",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "air-jordan",
    },
    {
      ID: 70898,
      post_author: "1",
      post_date: "2019-06-18 09:41:06",
      post_date_gmt: "2019-06-18 08:41:06",
      post_content: " ",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "70898",
      to_ping: "",
      pinged: "",
      post_modified: "2019-06-18 09:42:40",
      post_modified_gmt: "2019-06-18 08:42:40",
      post_content_filtered: "",
      post_parent: 0,
      guid: "https://fastsole.co.uk/?p=70898",
      menu_order: 4,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 70898,
      menu_item_parent: "0",
      object_id: "15",
      object: "post_tag",
      type: "taxonomy",
      type_label: "Tag",
      url: "/sneaker-news/tag/air-max/",
      title: "Air max",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "air-max",
    },
    {
      ID: 70899,
      post_author: "1",
      post_date: "2019-06-18 09:41:06",
      post_date_gmt: "2019-06-18 08:41:06",
      post_content: " ",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "70899",
      to_ping: "",
      pinged: "",
      post_modified: "2019-06-18 09:42:40",
      post_modified_gmt: "2019-06-18 08:42:40",
      post_content_filtered: "",
      post_parent: 0,
      guid: "https://fastsole.co.uk/?p=70899",
      menu_order: 5,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 70899,
      menu_item_parent: "0",
      object_id: "340",
      object: "post_tag",
      type: "taxonomy",
      type_label: "Tag",
      url: "/sneaker-news/tag/yeezy-boost/",
      title: "Yeezy Boost",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "yeezy-boost",
    },
    {
      ID: 70900,
      post_author: "1",
      post_date: "2019-06-18 09:41:06",
      post_date_gmt: "2019-06-18 08:41:06",
      post_content: " ",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "70900",
      to_ping: "",
      pinged: "",
      post_modified: "2019-06-18 09:42:40",
      post_modified_gmt: "2019-06-18 08:42:40",
      post_content_filtered: "",
      post_parent: 0,
      guid: "https://fastsole.co.uk/?p=70900",
      menu_order: 6,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 70900,
      menu_item_parent: "0",
      object_id: "23355",
      object: "post_tag",
      type: "taxonomy",
      type_label: "Tag",
      url: "/sneaker-news/tag/air-vapormax/",
      title: "Air VaporMax",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "air-vapormax",
    },
    {
      ID: 70903,
      post_author: "1",
      post_date: "2019-06-18 09:42:40",
      post_date_gmt: "2019-06-18 08:42:40",
      post_content: " ",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "70903",
      to_ping: "",
      pinged: "",
      post_modified: "2019-06-18 09:42:40",
      post_modified_gmt: "2019-06-18 08:42:40",
      post_content_filtered: "",
      post_parent: 0,
      guid: "https://fastsole.co.uk/?p=70903",
      menu_order: 7,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 70903,
      menu_item_parent: "0",
      object_id: "80",
      object: "post_tag",
      type: "taxonomy",
      type_label: "Tag",
      url: "/sneaker-news/tag/air-force/",
      title: "Air Force",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "air-force",
    },
  ],
};
