// @frontity/.ignored_mars-theme/src/model/FooterSupportModel.js
export const FooterSupportModel = {
  term_id: 226,
  name: "About",
  slug: "about",
  term_group: 0,
  term_taxonomy_id: 226,
  taxonomy: "nav_menu",
  description: "",
  parent: 0,
  count: 5,
  filter: "raw",
  items: [
    {
      ID: 6204,
      post_author: "1",
      post_date: "2017-01-31 17:56:35",
      post_date_gmt: "2017-01-31 17:56:35",
      post_content: " ",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "6204",
      to_ping: "",
      pinged: "",
      post_modified: "2021-11-17 10:31:00",
      post_modified_gmt: "2021-11-17 10:31:00",
      post_content_filtered: "",
      post_parent: 0,
      guid: "http://fastsole.co.uk/?p=6204",
      menu_order: 1,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 6204,
      menu_item_parent: "0",
      object_id: "2254",
      object: "page",
      type: "post_type",
      type_label: "Page",
      url: "/about/",
      title: "About Us",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "about",
    },
    {
      ID: 62150,
      post_author: "1",
      post_date: "2019-03-07 07:41:49",
      post_date_gmt: "2019-03-07 07:41:49",
      post_content: "",
      post_title: "Contact Us",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "contact-us-2",
      to_ping: "",
      pinged: "",
      post_modified: "2021-11-17 10:31:00",
      post_modified_gmt: "2021-11-17 10:31:00",
      post_content_filtered: "",
      post_parent: 0,
      guid: "https://fastsole.co.uk/?p=62150",
      menu_order: 2,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 62150,
      menu_item_parent: "0",
      object_id: "62144",
      object: "page",
      type: "post_type",
      type_label: "Page",
      url: "/contact/",
      title: "Contact Us",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "contact",
    },
    {
      ID: 56606,
      post_author: "1",
      post_date: "2019-01-14 14:39:09",
      post_date_gmt: "2019-01-14 14:39:09",
      post_content: "",
      post_title: "Terms",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "terms-2",
      to_ping: "",
      pinged: "",
      post_modified: "2021-11-17 10:31:00",
      post_modified_gmt: "2021-11-17 10:31:00",
      post_content_filtered: "",
      post_parent: 0,
      guid: "https://fastsole.co.uk/?p=56606",
      menu_order: 3,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 56606,
      menu_item_parent: "0",
      object_id: "56349",
      object: "page",
      type: "post_type",
      type_label: "Page",
      url: "/terms-conditions/",
      title: "Terms",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "terms-conditions",
    },
    {
      ID: 56607,
      post_author: "1",
      post_date: "2019-01-14 14:39:09",
      post_date_gmt: "2019-01-14 14:39:09",
      post_content: "",
      post_title: "Privacy",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "privacy-2",
      to_ping: "",
      pinged: "",
      post_modified: "2021-11-17 10:31:00",
      post_modified_gmt: "2021-11-17 10:31:00",
      post_content_filtered: "",
      post_parent: 0,
      guid: "https://fastsole.co.uk/?p=56607",
      menu_order: 4,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 56607,
      menu_item_parent: "0",
      object_id: "56343",
      object: "page",
      type: "post_type",
      type_label: "Page",
      url: "/privacy-policy/",
      title: "Privacy",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "privacy-policy",
    },
    /* {
      ID: 156114,
      post_author: "30",
      post_date: "2021-11-17 10:31:00",
      post_date_gmt: "2021-11-17 10:31:00",
      post_content: " ",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "156114",
      to_ping: "",
      pinged: "",
      post_modified: "2021-11-17 10:31:00",
      post_modified_gmt: "2021-11-17 10:31:00",
      post_content_filtered: "",
      post_parent: 0,
      guid: "https://fastsole.thenwg.xyz/?p=156114",
      menu_order: 5,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 156114,
      menu_item_parent: "0",
      object_id: "3948",
      object: "page",
      type: "post_type",
      type_label: "Page",
      url: "/sneaker-news/category/offer-discount/",
      title: "Black Friday 2016",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "black-friday-2016",
    }, */
  ],
};
