// @frontity/.ignored_mars-theme/src/model/FooterQuickLinkModel.js
export const FooterQuickLinkModel = {
  term_id: 231,
  name: "Quick Links",
  slug: "quick-links",
  term_group: 0,
  term_taxonomy_id: 231,
  taxonomy: "nav_menu",
  description: "",
  parent: 0,
  count: 7,
  filter: "raw",
  items: [
    {
      ID: 63786,
      post_author: "1",
      post_date: "2019-03-31 09:54:12",
      post_date_gmt: "2019-03-31 08:54:12",
      post_content:
        "BUY AIR JORDAN SHOES FOR MEN THE HISTORY OF NIKE AND AIR JORDAN SHOES Air Jordan is a brand name of basketball shoes manufactured by Nike. It was created for former professional basketball player Michael Jordan. The original Air Jordan I sneakers was produced exclusively for Jordan in early 1984 and released to the public in late 1984. The shoes were designed for Nike by Peter Moore, Tinker Hatfield, and Bruce Kilgore. Many people ignore the significance of buying good shoes. If your shoe size is wrong, then your feet will probably get cramp quickly, you won’t be able to manage your balance. To make sure no one has got ever to undergo this struggle. We at Fastsole has a wide selection of Air Jordan shoes in the UK that are eye-catching, comfortably built in, give exceptional shock absorption and therefore are all over excellent. The range of Air Jordan shoes online has various surface finishes and designs to match your preference. The wide variety assures you don’t need to stay for anything at all less than best sneakers. Air Jordan shoes made for the basketball game. These athletic shoes can also be used even though playing other sports activities.…",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "63786",
      to_ping: "",
      pinged: "",
      post_modified: "2019-03-31 09:54:18",
      post_modified_gmt: "2019-03-31 08:54:18",
      post_content_filtered: "",
      post_parent: 54,
      guid: "https://fastsole.co.uk/?p=63786",
      menu_order: 1,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 63786,
      menu_item_parent: "0",
      object_id: "46",
      object: "department",
      type: "taxonomy",
      type_label: "Department",
      url: "/sneaker-release-dates/brands/nike-jordan/",
      title: "Nike Air Jordan",
      target: "",
      attr_title: "",
      description:
        "BUY AIR JORDAN SHOES FOR MEN THE HISTORY OF NIKE AND AIR JORDAN SHOES Air Jordan is a brand name of basketball shoes manufactured by Nike. It was created for former professional basketball player Michael Jordan. The original Air Jordan I sneakers was produced exclusively for Jordan in early 1984 and released to the public in late 1984. The shoes were designed for Nike by Peter Moore, Tinker Hatfield, and Bruce Kilgore. Many people ignore the significance of buying good shoes. If your shoe size is wrong, then your feet will probably get cramp quickly, you won’t be able to manage your balance. To make sure no one has got ever to undergo this struggle. We at Fastsole has a wide selection of Air Jordan shoes in the UK that are eye-catching, comfortably built in, give exceptional shock absorption and therefore are all over excellent. The range of Air Jordan shoes online has various surface finishes and designs to match your preference. The wide variety assures you don’t need to stay for anything at all less than best sneakers. Air Jordan shoes made for the basketball game. These athletic shoes can also be used even though playing other sports activities.…",
      classes: [""],
      xfn: "",
      slug: "nike-jordan",
    },
    {
      ID: 63782,
      post_author: "1",
      post_date: "2019-03-31 09:54:12",
      post_date_gmt: "2019-03-31 08:54:12",
      post_content:
        "Nike is the world’s largest manufacturer of top quality sports shoes. They have taken over the market through exceptional performance products and trendy goods. An honest testimony of the factors is Nike Air Max Shoes who have confirmed to be a game-changing design from the brand which is favorite among athletics all around the world. Buy Nike Air Max Shoes Online and Experience the Comfort and ease while Running. It is time to say bye bye to all reasons and start giving your full effort towards taking on healthy living. Get ready to enjoy comfort and ease as you walk, run or do jogging with Nike Air Max shoes. The wide range of Air Max shoes from Nike has less middle of the sole material and greater size airbags to get more impact protection. Don’t be worried about standing in long queues to get a set when you could buy Nike Air Max shoes online. The Nike sneakers prices online have attractive deals. Nike Air Max Shoes for Men Built to be in light-weight on the soles, the Nike Air Max shoes for men undoubtedly are a set to be love. These sneakers are made utilizing larger-volume airbags, which assures your…",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "63782",
      to_ping: "",
      pinged: "",
      post_modified: "2019-03-31 09:54:18",
      post_modified_gmt: "2019-03-31 08:54:18",
      post_content_filtered: "",
      post_parent: 54,
      guid: "https://fastsole.co.uk/?p=63782",
      menu_order: 2,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 63782,
      menu_item_parent: "0",
      object_id: "622",
      object: "department",
      type: "taxonomy",
      type_label: "Department",
      url: "/sneaker-release-dates/brands/air-max/",
      title: "Air Max",
      target: "",
      attr_title: "",
      description:
        "Nike is the world’s largest manufacturer of top quality sports shoes. They have taken over the market through exceptional performance products and trendy goods. An honest testimony of the factors is Nike Air Max Shoes who have confirmed to be a game-changing design from the brand which is favorite among athletics all around the world. Buy Nike Air Max Shoes Online and Experience the Comfort and ease while Running. It is time to say bye bye to all reasons and start giving your full effort towards taking on healthy living. Get ready to enjoy comfort and ease as you walk, run or do jogging with Nike Air Max shoes. The wide range of Air Max shoes from Nike has less middle of the sole material and greater size airbags to get more impact protection. Don’t be worried about standing in long queues to get a set when you could buy Nike Air Max shoes online. The Nike sneakers prices online have attractive deals. Nike Air Max Shoes for Men Built to be in light-weight on the soles, the Nike Air Max shoes for men undoubtedly are a set to be love. These sneakers are made utilizing larger-volume airbags, which assures your…",
      classes: [""],
      xfn: "",
      slug: "air-max",
    },
    {
      ID: 63783,
      post_author: "1",
      post_date: "2019-03-31 09:54:12",
      post_date_gmt: "2019-03-31 08:54:12",
      post_content: " ",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "63783",
      to_ping: "",
      pinged: "",
      post_modified: "2019-03-31 09:54:18",
      post_modified_gmt: "2019-03-31 08:54:18",
      post_content_filtered: "",
      post_parent: 622,
      guid: "https://fastsole.co.uk/?p=63783",
      menu_order: 3,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 63783,
      menu_item_parent: "0",
      object_id: "26890",
      object: "department",
      type: "taxonomy",
      type_label: "Department",
      url: "/sneaker-release-dates/brands/air-max-720/",
      title: "Air Max 720",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "air-max-720",
    },
    {
      ID: 63784,
      post_author: "1",
      post_date: "2019-03-31 09:54:12",
      post_date_gmt: "2019-03-31 08:54:12",
      post_content: " ",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "63784",
      to_ping: "",
      pinged: "",
      post_modified: "2019-03-31 09:54:18",
      post_modified_gmt: "2019-03-31 08:54:18",
      post_content_filtered: "",
      post_parent: 622,
      guid: "https://fastsole.co.uk/?p=63784",
      menu_order: 4,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 63784,
      menu_item_parent: "0",
      object_id: "27106",
      object: "department",
      type: "taxonomy",
      type_label: "Department",
      url: "/sneaker-release-dates/brands/air-max-95/",
      title: "Air Max 95",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "air-max-95",
    },
    {
      ID: 63785,
      post_author: "1",
      post_date: "2019-03-31 09:54:12",
      post_date_gmt: "2019-03-31 08:54:12",
      post_content: " ",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "63785",
      to_ping: "",
      pinged: "",
      post_modified: "2019-03-31 09:54:18",
      post_modified_gmt: "2019-03-31 08:54:18",
      post_content_filtered: "",
      post_parent: 622,
      guid: "https://fastsole.co.uk/?p=63785",
      menu_order: 5,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 63785,
      menu_item_parent: "0",
      object_id: "27107",
      object: "department",
      type: "taxonomy",
      type_label: "Department",
      url: "/sneaker-release-dates/brands/air-max-97/",
      title: "Air Max 97",
      target: "",
      attr_title: "",
      description: "",
      classes: [""],
      xfn: "",
      slug: "air-max-97",
    },
    {
      ID: 63787,
      post_author: "1",
      post_date: "2019-03-31 09:54:12",
      post_date_gmt: "2019-03-31 08:54:12",
      post_content:
        "Buy Nike Air VaporMax Shoes for Men and Women Feel the upcoming emerging trend in unique Air technological innovation together with the significantly reinvented Air shock absorption technique made by the Nike Air VaporMax. Repel common pondering kind. The Nike VaporMax shoes offer a surprisingly light-weight, flexible and versatile experience with obvious cross-over attraction. Go through the whole perspective of Nike Air technology and rejoice more than three decades of footwear creativity with a pair of revolutionary Air VaporMax shoes. Set within the junction of technology, performance and design styles, the Air VaporMax presents a breakthrough to the Nike Air Max family. New technological enhancements brought about the elimination of an old-fashioned foam midsole, enabling Nike Air technology in the future much better the foot than before. This process crafted a standalone outsole which provides one of the most direct experiences with Air technology up to now. The VaporMax Air unit brings together using the resilient Nike Flyknit upper, providing a sneaker with streets attraction, great flexibleness and responsiveness. The multipurpose sneaker is well-suited for your run with light-weight cushion padding for comfort and eases even long distance. Its smooth design and style apply very easily to day-to-day dress in,…",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "63787",
      to_ping: "",
      pinged: "",
      post_modified: "2019-03-31 09:54:18",
      post_modified_gmt: "2019-03-31 08:54:18",
      post_content_filtered: "",
      post_parent: 54,
      guid: "https://fastsole.co.uk/?p=63787",
      menu_order: 6,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 63787,
      menu_item_parent: "0",
      object_id: "2669",
      object: "department",
      type: "taxonomy",
      type_label: "Department",
      url: "/sneaker-release-dates/brands/air-vapormax/",
      title: "Air VaporMax",
      target: "",
      attr_title: "",
      description:
        "Buy Nike Air VaporMax Shoes for Men and Women Feel the upcoming emerging trend in unique Air technological innovation together with the significantly reinvented Air shock absorption technique made by the Nike Air VaporMax. Repel common pondering kind. The Nike VaporMax shoes offer a surprisingly light-weight, flexible and versatile experience with obvious cross-over attraction. Go through the whole perspective of Nike Air technology and rejoice more than three decades of footwear creativity with a pair of revolutionary Air VaporMax shoes. Set within the junction of technology, performance and design styles, the Air VaporMax presents a breakthrough to the Nike Air Max family. New technological enhancements brought about the elimination of an old-fashioned foam midsole, enabling Nike Air technology in the future much better the foot than before. This process crafted a standalone outsole which provides one of the most direct experiences with Air technology up to now. The VaporMax Air unit brings together using the resilient Nike Flyknit upper, providing a sneaker with streets attraction, great flexibleness and responsiveness. The multipurpose sneaker is well-suited for your run with light-weight cushion padding for comfort and eases even long distance. Its smooth design and style apply very easily to day-to-day dress in,…",
      classes: [""],
      xfn: "",
      slug: "air-vapormax",
    },
    {
      ID: 56611,
      post_author: "1",
      post_date: "2019-01-14 14:44:07",
      post_date_gmt: "2019-01-14 14:44:07",
      post_content:
        "FastSole is the home of Men’s Designer Casual Shoes. Among them out there Yeezy is one of the name for the greatest footwear experience. Are you looking Yeezy  release dates and news in uk? look no further because Fastsole is the right place where we provide you release date and upcoming news of Yeezy shoes  in UK and Europe.We provide all the detailed info on the rumours around the internet.  As we prepare for upcoming holidays, we will update you with all the confirmed news on difference colourways of Yeezy Boost 350 V2 Trainers. Yeezy Sneaker Release Dates in UK | Yeezy Launch Date and News in UK",
      post_title: "",
      post_excerpt: "",
      post_status: "publish",
      comment_status: "closed",
      ping_status: "closed",
      post_password: "",
      post_name: "56611",
      to_ping: "",
      pinged: "",
      post_modified: "2019-03-31 09:54:18",
      post_modified_gmt: "2019-03-31 08:54:18",
      post_content_filtered: "",
      post_parent: 50,
      guid: "https://fastsole.co.uk/?p=56611",
      menu_order: 7,
      post_type: "nav_menu_item",
      post_mime_type: "",
      comment_count: "0",
      filter: "raw",
      db_id: 56611,
      menu_item_parent: "0",
      object_id: "148",
      object: "department",
      type: "taxonomy",
      type_label: "Department",
      url: "/sneaker-release-dates/brands/yeezy/",
      title: "Yeezy",
      target: "",
      attr_title: "",
      description:
        "FastSole is the home of Men’s Designer Casual Shoes. Among them out there Yeezy is one of the name for the greatest footwear experience. Are you looking Yeezy  release dates and news in uk? look no further because Fastsole is the right place where we provide you release date and upcoming news of Yeezy shoes  in UK and Europe.We provide all the detailed info on the rumours around the internet.  As we prepare for upcoming holidays, we will update you with all the confirmed news on difference colourways of Yeezy Boost 350 V2 Trainers. Yeezy Sneaker Release Dates in UK | Yeezy Launch Date and News in UK",
      classes: [""],
      xfn: "",
      slug: "yeezy",
    },
  ],
};
